futaba/cogs/moderation/core.py

```diff
@@ -439,3 +439,41 @@ async def unban(self, ctx, user: UserConv, *, reason: str):
             raise ManualCheckFailure(
                 content="I don't have permission to unban this user"
             )
+
+    @commands.command(name="msgcollapse", aliases=["gist"])
+    @commands.guild_only()
+    async def msgcollapse(self, ctx, *messages: MessageConv):
+        """
+        Concatenates the range of messages and uploads to a gist.
+        The original messages are deleted and a link to the gist is posted.
+
+        Note: The messages specified should be by the same user
+        """
+
+        if not self.bot.config.gist_oauth_token:
+            raise CommandFailed(content="The gist oauth token is not configured.")
+
+        if not permissions.has_perm(ctx, "manage_messages") and any(
+            message.author.id != ctx.author.id for message in messages
+        ):
+            # check if the messages were created by the same user
+            raise ManualCheckFailure(content="I can only collapse your messages")
+
+        logger.info("Collapsing %d messages into a gist", len(messages))
+
+        gist_url = await gist.create_single_gist(
+            token=self.bot.config.gist_oauth_token,
+            content="  \n".join(str(message.content) for message in messages),
+            filename="collapsed.md",
+            description="Discord collapsed messages",
+            public=False,
+        )
```

Comment:
As mentioned above, I would recommend a generic method on `bot` that fills in the arguments that are common

Reply:
So should I add these & token into settings(cog & sql)? Or maybe move the whole functionality into like an optional cog so that it's distinct from moderation, and would use the `optional_cog_settings` table?

Reply:
This can be in the main cog I think, I'm just suggesting you have a helper method for creating gists. This way it's easier for other parts of the bot in the future to also use it without violating DRY.

Reply:
Oh ok, I just thought that there might be an issue with storing the github token in the main config because the bot might be used by multiple guilds.

Reply:
Is it okay that the helper method is also a command? Because otherwise it would just be a passthrough method.

Reply:
futaba is only in 3 guilds, however having the allowing a github token per guild would be fine, lets the guild manage the gists created, a help method was just suggested in case we want other thing to be added to a gist that way it can be managed in one place